tracer/build/_build/Build.VariableGenerations.cs

```diff
@@ -0,0 +1,263 @@
+                {
+                    Logger.Info("Filter condition is not provided, assuming true.");
+                    return true;
+                }
+
+                var varFilterConditions = GenerateConditionVariableFilter.Split(';');
+
+                var isChanged = false;
+
+                foreach (var varFilterCondition in varFilterConditions)
+                {
+                    Logger.Info($"Checking changes for {varFilterCondition}");
+                    isChanged = CheckChange(varFilterCondition) || isChanged;
+                }
+
+                return isChanged;
+
+                bool CheckChange(string varFilterCondition)
+                {
+                    var arr = varFilterCondition.Split('|');
+                    var variableName = arr[0];
+                    var filters = arr[1].Split(',');
+
+                    var masterCommit = GitTasks.Git("merge-base origin/master HEAD").First().Text;
+                    var changedFiles =
+                            GitTasks
+                               .Git($"diff --name-only \"{masterCommit}\"")
+                               .Select(output => output.Text)
+                               .ToArray()
+                        ;
+
+                    var isChanged = filters.Any(filter => changedFiles.Any(s => s.Contains(filter)));
+
+                    Logger.Info($"{variableName} - {isChanged}");
+
+                    var variableValue = isChanged.ToString();
+                    EnvironmentInfo.SetVariable(variableName, variableValue);
+                    AzurePipelines.Instance.SetVariable(variableName, variableValue);
+
+                    return isChanged;
+                }
+            }
+
+            void GenerateIntegrationTestsWindowsMatrices(GenerateMatricesTarget target)
+            {
+                var targetFrameworks = TargetFramework.GetFrameworks(new[] { TargetFramework.NETSTANDARD2_0 });
+
+                if (target.HasFlag(global::GenerateMatricesTarget.integration_tests_windows_matrix))
+                {
+                    GenerateIntegrationTestsWindowsMatrix(targetFrameworks);
+                }
+
+                if (target.HasFlag(global::GenerateMatricesTarget.integration_tests_windows_iis_matrix))
+                {
+                    GenerateIntegrationTestsWindowsIISMatrix(targetFrameworks);
+                }
+            }
+
+            void GenerateIntegrationTestsWindowsMatrix(TargetFramework[] targetFrameworks)
+            {
+                var targetPlatforms = new[] { "x86", "x64" };
+                var matrix = new Dictionary<string, object>();
+
+                foreach (var framework in targetFrameworks)
+                {
+                    foreach (var targetPlatform in targetPlatforms)
+                    {
+                        matrix.Add($"{targetPlatform}_{framework}", new { framework, targetPlatform });
+                    }
+                }
```

Comment:
Very much personal preference I think, but personally I would prefer query syntax for this. Either is fine though :)
```suggestion
                var matrix = (from framework in targetFrameworks
                              from targetPlatform in new [] { "x86", "x64" }
                              select ($"{targetPlatform}_{framework}", new { framework, targetPlatform }))
                              .ToDictionary(x=>x.Item1, x=>x.Item2);
```

Reply:
Totally agree, and I would suggest doing the same in `GenerateExplorationTestsLinuxMatrix`. I admit that it's nitpicky and comes down to personal preference of course 🙂, but one could argue that this is the idiomatic way to express you want a Cartesian Product in C#.

Reply:
I prefer to leave it as is.
In this way, it could be easily extended by custom conditions.
You can see linux exploration tests for example.

Reply:
>In this way, it could be easily extended by custom conditions

FWIW, this format can handle conditionals just fine. e.g. for `GenerateExplorationTestsLinuxMatrix`:

```csharp
var matrix = (from baseImage in new[] { "debian", "alpine" }
              from explorationTestUseCase in useCases
              from targetFramework in new [] { "x86", "x64" }
              from testDescription in testDescriptions
              where testDescription.IsFrameworkSupported(targetFramework))
              select ($"{baseImage}_{targetFramework}_{explorationTestUseCase}_{testDescription.Name}",
                                                      new { baseImage, targetFramework, explorationTestUseCase, explorationTestName = testDescription.Name }))
              .ToDictionary(x=>x.Item1, x=>x.Item2);
```